### PR TITLE
Upgrade issue templates

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
## Description

Upgrade to GitHub issue forms from old issue templates.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

**Comment by krobelus** [comment]
Thanks for the thought but we've previously rejected issue templates in https://github.com/fish-shell/fish-shell/issues/7394
I don't think we have a real need to make our process more bureaucratic.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-Please tell us which fish version you are using by executing the following:
-
-  fish --version
-  echo $version
-
-Please tell us which operating system and terminal you are using. The output of `uname -a` and `echo $TERM` may be helpful in this regard although other commands might be relevant in your specific situation.
-
-Please tell us if you tried fish without third-party customizations by executing this command and whether it affected the behavior you are reporting:
-
-  sh -c 'env HOME=$(mktemp -d) fish'
```
Comment (krobelus) [inline .github/ISSUE_TEMPLATE.md:11]: this command is very important because it solves most user problems

**Comment by krobelus** [comment]
Duplicate of #7394

**Review by faho — CHANGES_REQUESTED** [review]
You know, it would be fantastic if github had a way to testdrive this. I don't think they do.

In general, I don't believe categorizing issues immediately makes too much sense - many many many feature requests we have gotten started out as bugs simply because people didn't know it didn't exist.

And this seems to ignore questions, which are a good ~40% of our issues.

I'm not sold on this, tbh.

Some minor suggestions in-line.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: true
```
[the remark this answers is not in the document]
Comment (EmilySeville7cfg) [reply .github/ISSUE_TEMPLATE/config.yml:1]: Yes, blank issues are for questions. ;)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    attributes:
+      label: Steps to reproduce
+      description: >
+        Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include code to reproduce, if relevant.
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Fish version
+      placeholder: "How to get: fish --version"
```
[the remark this answers is not in the document]
Comment (EmilySeville7cfg) [reply .github/ISSUE_TEMPLATE/bug_report.yml:42]: What about to restrict bugs/feature requests creating for all versions below `3.2.0`?

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (EmilySeville7cfg) [reply .github/ISSUE_TEMPLATE.md:11]: It's restored. Thanks!

**Pull request closed without merging**